src/img/gen-color-table.pl

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/perl -w
+
+# Generates xpm-color-table.h from X11's rgb.txt
+# Adapted from gdk-pixbuf (LGPL-2.0-or-later)
+
+if (@ARGV != 1) {
+    die "Usage: gen-color-table.pl rgb.txt > xpm-color-table.h\n";
+}
+
+open IN, $ARGV[0] || die "Cannot open $ARGV[0]: $!\n";
+
+@colors = ();
+while (defined($_ = <IN>)) {
```

Comment:
there is always newline (or last line w/o ! ... )so while (<IN>) would suffice, but if this is style somewhere so be it :D

Reply:
This whole script was pretty much copied from gdk-pixbuf. I know very little perl - would you care to post a diff of how to improve it?

Alternatively, we could just drop the script entirely - the X11 color definitions have not changed in over a decade and are unlikely to ever again at this point, I expect.

Reply:
So you mean save the generated code ?
(I was wondering how is this used, running at build time? -- or just run once and use the generated code.)
-- for the case this script is going to be preserved in repo, I'll paste diff here:

```
--- gen-color-table.pl-orig	2024-09-30 21:45:22.134618483 +0300
+++ gen-color-table.pl	2024-09-30 21:47:51.115934367 +0300
@@ -7,18 +7,20 @@
     die "Usage: gen-color-table.pl rgb.txt > xpm-color-table.h\n";
 }
 
-open IN, $ARGV[0] || die "Cannot open $ARGV[0]: $!\n";
+open IN, '<', $ARGV[0] or die "Cannot open $ARGV[0]: $!\n";
 
 @colors = ();
-while (defined($_ = <IN>)) {
+while (<IN>) {
     next if /^!/;
-    if (!/^\s*([0-9]+)\s+([0-9]+)\s+([0-9]+)\s+(.*\S)\s+$/) {
+    if (!/^\s*([0-9]+)\s+([0-9]+)\s+([0-9]+)\s+(.*\S)\s*$/) {
 	die "Cannot parse line $_";
     }
 
     push @colors, [$1, $2, $3, $4];
 }
 
+close IN or die "close IN failed: $!\n";
+
 @colors = sort { lc($a->[3]) cmp lc($b->[3]) } @colors;
 
 $offset = 0;
```

Reply:
I added these fixes to the PR, thanks :)